service/src/main/java/com/myhome/security/filters/CommunityAuthorizationFilter.java

```diff
@@ -0,0 +1,66 @@
+package com.myhome.security.filters;
+
+import com.myhome.domain.User;
+import com.myhome.services.CommunityService;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CommunityAuthorizationFilter extends BasicAuthenticationFilter {
+    private final CommunityService communityService;
+    private final String uuidPattern = "[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}";
+    private final Pattern addAmenityRequestPattern = Pattern.compile("/communities/" + uuidPattern + "/amenities");
+
+
+    public CommunityAuthorizationFilter(AuthenticationManager authenticationManager,
+                                        CommunityService communityService) {
+        super(authenticationManager);
+        this.communityService = communityService;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response,
+                                    FilterChain chain) throws IOException, ServletException {
+
+        Matcher urlMatcher = addAmenityRequestPattern.matcher(request.getRequestURI());
+
+        if (urlMatcher.find() && !isUserCommunityAdmin(request)) {
+            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
```

Comment:
It should be 403 instead of 401. 

Authorization - 403 - what user can have access to
Authentication - 401 - who a user is